src/connect.js

```diff
@@ -20,7 +20,12 @@
  * @return {Function}
  */
 function connect(params) {
-  // write code here
+  const baseParams = params;
+  let overalParam = {};
+  return function(extraParams) {
```

Comment:
you can write simply 
```suggestion
  return (extraParams) => { ...params, ... extraParams };
```